client/src/modules/dashboard/shared/components/RoleChangeModal.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, Select, Text, UseDisclosureReturn } from '@chakra-ui/react';
+import React from 'react';
+import { useForm } from 'react-hook-form';
+
+import { Modal } from '../../../../components/Modal';
+
+export interface RoleChangeModalData {
+  roleId: number;
+  userId: number;
+  userName: string;
+}
+
+export const RoleChangeModal: React.FC<{
+  modalProps: UseDisclosureReturn;
+  data: RoleChangeModalData;
+  roles: { id: number; name: string }[];
+  title: string;
+  onSubmit: (submitData: {
+    newRoleId: number;
+    userId: number;
+  }) => Promise<void>;
+}> = ({ modalProps, data, roles, title, onSubmit }) => {
+  const { handleSubmit, register, setValue } = useForm<{
+    newRoleId: number;
+    userId: number;
+  }>();
+
+  setValue('userId', data.userId);
```

Comment:
Silly question, probably, but what does `setValue` achieve over passing in `defaultValues`? If it's essentially the same, I'd recommend using `defaultValues` for consistency.

Reply:
Some questions can be explained with me not being too versed with react related things.

I think ideally would be to not need passing `userId` to the component at all, as it doesn't do anything there. I did that because I was having some issues with _Object is possibly 'undefined'._ when trying to use `chapterUser.userId` in the `onModalSubmit`.
```ts
    changeRoleMutation({
      variables: {
        chapterId: id,
        roleId: data.newRoleId,
        userId: chapterUser.userId,
      },
    });
```

When checking using `defaultValues`:
```ts
  const { handleSubmit, register } = useForm<{
    newRoleId: number;
    userId: number;
  }>({ defaultValues: { userId: data.userId } });
```
Role is changed only for the single user - for whom modal was open first. Opening modal for different user will still change role for the first one.

---
Looking at this exposed bug that I didn't notice before - the selected role in the modal doesn't change when opening it for different users. So I'm going to look at it some more.

Reply:
Ah, thanks for the explanation, that's some strange behaviour from react-hook-form.